tests/lib/linter/node-event-generator.js

```diff
@@ -1,565 +0,0 @@
-// Requirements
-//------------------------------------------------------------------------------
-
-const assert = require("node:assert"),
-	sinon = require("sinon"),
-	espree = require("espree"),
-	vk = require("eslint-visitor-keys"),
-	Traverser = require("../../../lib/shared/traverser"),
-	EventGeneratorTester = require("../../../tools/internal-testers/event-generator-tester"),
-	createEmitter = require("../../../lib/linter/safe-emitter"),
-	NodeEventGenerator = require("../../../lib/linter/node-event-generator");
-
-//------------------------------------------------------------------------------
-// Constants
-//------------------------------------------------------------------------------
-
-const ESPREE_CONFIG = {
-	ecmaVersion: 6,
-	comment: true,
-	tokens: true,
-	range: true,
-	loc: true,
-};
-
-const STANDARD_ESQUERY_OPTION = {
-	visitorKeys: vk.KEYS,
-	fallback: Traverser.getKeys,
-};
-
-//------------------------------------------------------------------------------
-// Tests
-//------------------------------------------------------------------------------
-
-describe("NodeEventGenerator", () => {
-	EventGeneratorTester.testEventGeneratorInterface(
-		new NodeEventGenerator(createEmitter(), STANDARD_ESQUERY_OPTION),
-	);
-
-	describe("entering a single AST node", () => {
-		let emitter, generator;
-
-		beforeEach(() => {
-			emitter = Object.create(createEmitter(), {
-				emit: { value: sinon.spy() },
-			});
-
-			["Foo", "Bar", "Foo > Bar", "Foo:exit"].forEach(selector =>
-				emitter.on(selector, () => {}),
-			);
-			generator = new NodeEventGenerator(
-				emitter,
-				STANDARD_ESQUERY_OPTION,
-			);
-		});
-
-		it("should generate events for entering AST node.", () => {
-			const dummyNode = { type: "Foo", value: 1 };
-
-			generator.enterNode(dummyNode);
-
-			assert(emitter.emit.calledOnce);
-			assert(emitter.emit.calledWith("Foo", dummyNode));
-		});
-
-		it("should generate events for exiting AST node.", () => {
-			const dummyNode = { type: "Foo", value: 1 };
-
-			generator.leaveNode(dummyNode);
-
-			assert(emitter.emit.calledOnce);
-			assert(emitter.emit.calledWith("Foo:exit", dummyNode));
-		});
-
-		it("should generate events for AST queries", () => {
-			const dummyParent = { type: "Foo" };
-			const dummyNode = { type: "Bar" };
-
-			generator.enterNode(dummyParent);
-			generator.enterNode(dummyNode);
-
-			assert(emitter.emit.calledThrice);
-			assert(emitter.emit.calledWith("Foo", dummyParent));
-			assert(emitter.emit.calledWith("Foo > Bar", dummyNode));
-			assert(emitter.emit.calledWith("Bar", dummyNode));
-		});
-
-		it("should use `nodeTypeKey` if provided", () => {
-			generator = new NodeEventGenerator(emitter, {
-				...STANDARD_ESQUERY_OPTION,
-				nodeTypeKey: "kind",
-			});
-
-			const dummyNode = { kind: "Foo" };
-
-			generator.enterNode(dummyNode);
-
-			assert(emitter.emit.calledOnce);
-			assert(emitter.emit.calledWith("Foo", dummyNode));
-		});
```

Comment:
I think an equivalent of this test is missing in the new test file. It's important to ensure we don't accidentally use hardcoded `node.type` without noticing, as was the case before https://github.com/eslint/eslint/commit/910bd13c4cb49001f2a9f172229360771b857585.